src/pages/users/UserInfoEditModal.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import axios from "axios";
+import { DateTime } from "luxon";
+import { PatternFormat } from "react-number-format";
+import { AxiosRefetch } from "../../util/types";
+import { CheckCircleIcon, WarningIcon, WarningTwoIcon } from "@chakra-ui/icons";
+
+const PROVIDER_ID_MAP: Record<any, string> = {
+  "google.com": "Google",
+  "github.com": "GitHub",
+  "saml.georgia-tech-login": "GT Login",
+  password: "Password",
+};
+
+interface Props {
+  defaultValues: Partial<any>;
+  isOpen: boolean;
+  onClose: () => void;
+  refetch: AxiosRefetch;
+}
+
+const UserInfoEditModal: React.FC<Props> = (props) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { isSubmitting },
+    reset,
+  } = useForm();
+
+  useEffect(() => {
+    reset(props.defaultValues);
+  }, [props.defaultValues, reset]);
+
+  const toast = useToast();
+
+  const handleFormSubmit = async (values: any) => {
+    try {
+      await axios.post(
+        `https://auth.api.hexlabs.org/permissions/${values.userId}`,
+        values
+      );
+      await props.refetch();
+      toast({
+        title: "Success!",
+        description: "Permissions have successfully been edited.",
+        status: "success",
+        duration: 3000,
+        isClosable: true,
+      });
+    } catch (e: any) {
+      toast({
+        title: "Error!",
+        description: "Could not edit. Please try again later.",
+        status: "error",
+        duration: 3000,
+        isClosable: true,
+      });
+    } finally {
+      props.onClose();
+    }
+  };
+
+  return (
+    <Modal onClose={props.onClose} isOpen={props.isOpen} isCentered>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader pb={1}>User Information</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <Stack mb="5">
+            <Box>
+              {props.defaultValues?.disabled && (
+                <Tag colorScheme="red" size="lg">
+                  <TagLeftIcon as={WarningTwoIcon} />
+                  <TagLabel>Account Disabled</TagLabel>
+                </Tag>
+              )}
+            </Box>
+            <Box>
+              <Text color="gray" fontSize="sm">
+                Name
+              </Text>
+              <Text>
+                {props.defaultValues?.name?.first}
+                {props.defaultValues?.name?.middle ? " " : ""}
+                {props.defaultValues?.name?.middle}{" "}
+                {props.defaultValues?.name?.last}
+              </Text>
+            </Box>
+            <Box>
+              <Text color="gray" fontSize="sm">
+                Email
+              </Text>
+              <Text display="inline" mr="2">
+                {props.defaultValues?.email}
+              </Text>
+              {props.defaultValues?.emailVerified ? (
+                <Tag>
+                  <TagLeftIcon as={CheckCircleIcon} />
+                  <TagLabel>Verified</TagLabel>
```

Comment:
looks great! might look better if the tag has a green background

Reply:
I was thinking about adding a green background, but I decided against it since it would attract too much attention. Since 99% of profiles will be verified, I thought it was a little distracting, and only thought to color it red if it's not verified.

Reply:
fair enough